frontend/routes/components/notifications/notifications.component.tsx

```diff
@@ -182,35 +260,15 @@ export class Notifications extends React.PureComponent<IProps, any> {
 		</>
 	)
 
-	public renderNotificationsHeader = () => (
+	private renderNotificationsHeader = () => (
 		<ListSubheaderToolbar rightContent={this.renderRightContent()}>
 			<Typography variant="title" color="inherit">
 				Notifications
 			</Typography>
 		</ListSubheaderToolbar>
 	)
 
-	public componentDidUpdate(prevProps: IProps) {
-		if (prevProps.notifications !== this.props.notifications) {
-			const unreadCount =  this.props.notifications.filter((n) => !n.read).length;
-			const groupedByTeamspace = { thisWeek: [], lastWeek: [], older: []};
-
-			const thisWeek = this.thisWeeksNotifications(this.props.notifications);
-			const lastWeek = this.lastWeeksNotifications(this.props.notifications);
-			const older = this.moreThanTwoWeeksAgoNotifications(this.props.notifications);
-
-			groupedByTeamspace.thisWeek = this.groupByTeamSpace(thisWeek);
-			groupedByTeamspace.lastWeek = this.groupByTeamSpace(lastWeek);
-			groupedByTeamspace.older = this.groupByTeamSpace(older);
-
-			const hasThisWeekNot = thisWeek.length > 0 ;
-			const hasLastWeekNot = lastWeek.length > 0;
-			const hasOlderNot = older.length > 0;
-			this.setState({unreadCount, groupedByTeamspace, hasThisWeekNot, hasLastWeekNot, hasOlderNot });
-		}
-	}
-
-	public renderNotificationsPanel = (data) => {
+	private renderNotificationsPanel = (data) => {
```

Comment:
why?

Reply:
I refactored the code - added some getters, set some methods as private, set proper order of fields and methods etc.